src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -238,31 +264,56 @@ val qualityFormat: Option[FastqQualityFormat] = None,
  @arg(doc="Predicted median insert size, to insert into the read group header") val predictedInsertSize: Option[Integer] = None,
  @arg(doc="Platform model to insert into the group header (ex. miseq, hiseq2500, hiseqX)") val platformModel: Option[String] = None,
  @arg(doc="Comment(s) to include in the merged output file's header.", minElements = 0) val comments: List[String] = Nil,
- @arg(doc="Date the run was produced, to insert into the read group header") val runDate: Option[Iso8601Date] = None
+ @arg(doc="Date the run was produced, to insert into the read group header") val runDate: Option[Iso8601Date] = None,
+ @arg(doc="Output FASTQs (.fastq.gz) in addition to BAM files") val outputFastqs: Boolean = false
 ) extends FgBioTool with LazyLogging {
 
   import DemuxFastqs._
 
+  private[fastq] val metricsPath = metrics.getOrElse(output.resolve(DefaultDemuxMetricsFileName))
+
   validate(inputs.length == readStructures.length, "The same number of read structures should be given as FASTQs.")
   validate(readStructures.flatMap(_.sampleBarcode).nonEmpty, s"No sample barcodes found in read structures: " + readStructures.map(_.toString).mkString(", "))
 
-  /** True if the read structure implies paired end reads will be produced, false otherwise. */
-  val pairedEnd: Boolean = readStructures.count(_.template.nonEmpty) match {
-    case 1 => false
-    case 2 => true
+  readStructures.count(_.template.nonEmpty) match {
```

Comment:
readStructures is a `Seq[ReadStructure]`, while `_.template` returns a list of all template segments (`Seq[ReadSegment]`) in that read structure, so we count how many read structures have at least one read segment that is a template.  I think it handles the case you're think about.